src/Workspaces/CoreTestUtilities/RenameHelpers.cs

```diff
@@ -0,0 +1,75 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Rename;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities.Utilities
+{
+    public static class RenameHelpers
+    {
+        public static async Task AssertRenameAnnotationsAsync(Solution originalSolution, Solution newSolution, IReadOnlyDictionary<string, string> symbolPairs, CancellationToken cancellationToken = default)
+        {
+            var allDocuments = newSolution.Projects.SelectMany(p => p.Documents);
+            var remainingSymbols = symbolPairs.ToHashSet();
+
+            foreach (var document in allDocuments)
+            {
+                var root = await document.GetSyntaxRootAsync(cancellationToken);
+                var annotatedNodesAndTokens = root.GetAnnotatedNodesAndTokens(RenameSymbolAnnotation.RenameSymbolKind);
+
+                var annotatedNodes = annotatedNodesAndTokens.Select(nodeOrToken => nodeOrToken.IsNode ? nodeOrToken.AsNode() : nodeOrToken.AsToken().Parent);
+
+                var originalDocument = originalSolution.GetDocument(document.Id);
+                var originalSemanticModel = await originalDocument.GetSemanticModelAsync(cancellationToken);
+                var originalCompilation = originalSemanticModel.Compilation;
+
+                var newSemanticModel = await document.GetSemanticModelAsync(cancellationToken);
+
+                foreach (var node in annotatedNodes)
+                {
+                    var annotation = node.GetAnnotations(RenameSymbolAnnotation.RenameSymbolKind).Single();
+                    var originalSymbol = RenameSymbolAnnotation.ResolveSymbol(annotation, originalCompilation);
+                    var newSymbol = newSemanticModel.GetDeclaredSymbol(node, cancellationToken);
+
+                    Assert.NotNull(originalSymbol);
+                    Assert.NotNull(newSymbol);
+
+                    var pair = (originalSymbol.ToDisplayString(), newSymbol.ToDisplayString());
+
+                    var isRemoved = remainingSymbols.Remove(pair.ToKeyValuePair());
+                    if (!isRemoved)
+                    {
+                        Assert.True(symbolPairs.Contains(pair.ToKeyValuePair()), $"Rename annotation for {pair} not expected");
+                    }
+                }
+            }
+
+            Assert.Empty(remainingSymbols);
+        }
+
+        public static ImmutableDictionary<string, string> MakeSymbolPairs(params (string, string)[] pairs)
+            => pairs.ToImmutableDictionary(p => p.Item1, p => p.Item2);
+
+        public static ImmutableDictionary<string, string> MakeSymbolPairs(params string[] items)
+        {
+            Assert.Equal(0, items.Length % 2);
```

Comment:
Consider adding a better explanation text here. If somebody mis-edits the test the failure will be "expected zero, actual 1" instead of "your pairs aren't paired up".